src/PoolTogetherGovernor.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.18;
+
+import {
+  Governor, GovernorCountingFractional
+} from "flexible-voting/src/GovernorCountingFractional.sol";
+import {IERC165} from "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {
+  ERC20VotesComp,
+  GovernorVotesComp
+} from "@openzeppelin/contracts/governance/extensions/GovernorVotesComp.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+import {ICompoundTimelock} from
+  "@openzeppelin/contracts/governance/extensions/GovernorTimelockCompound.sol";
+import {GovernorSettings} from "@openzeppelin/contracts/governance/extensions/GovernorSettings.sol";
+
+import {IPoolTogetherTimelock} from "src/interfaces/IPoolTogetherTimelock.sol";
+import {GovernorTimelockCompound} from "src/lib/GovernorTimelockCompound.sol";
+
+/// @notice The upgraded PoolTogether Governor: Bravo compatible and built with OpenZeppelin.
+contract PoolTogetherGovernor is
+  GovernorCountingFractional,
+  GovernorVotesComp,
+  GovernorTimelockCompound,
+  GovernorSettings
+{
+  /// @notice The address of the POOL token on Ethereum mainnet from which this Governor derives
+  /// delegated voting weight.
+  ERC20VotesComp private constant POOL_TOKEN =
+    ERC20VotesComp(0x0cEC1A9154Ff802e7934Fc916Ed7Ca50bDE6844e);
+
+  /// @notice The address of the existing PoolTogether DAO Timelock on Ethereum mainnet through
+  /// which
+  /// this Governor executes transactions.
+  IPoolTogetherTimelock private constant TIMELOCK =
+    IPoolTogetherTimelock(payable(0x42cd8312D2BCe04277dD5161832460e95b24262E));
+
+  /// @notice Human readable name of this Governor.
+  string private constant GOVERNOR_NAME = "PoolTogether Governor Bravo";
+
+  /// @notice The number of POOL (in "wei") that must participate in a vote for it to meet quorum
+  /// threshold.
+  ///
+  /// TODO: placeholder
+  uint256 private constant QUORUM = 100_000e18; // 100,000 POOL
+
+  /// @param _initialVotingDelay The deployment value for the voting delay this Governor will
+  /// enforce.
+  /// @param _initialVotingPeriod The deployment value for the voting period this Governor will
+  /// enforce.
+  /// @param _initialProposalThreshold The deployment value for the number of POOL required to
+  /// submit
+  /// a proposal this Governor will enforce.
+  constructor(
+    uint256 _initialVotingDelay,
+    uint256 _initialVotingPeriod,
+    uint256 _initialProposalThreshold
+  )
+    GovernorVotesComp(POOL_TOKEN)
+    GovernorSettings(_initialVotingDelay, _initialVotingPeriod, _initialProposalThreshold)
+    GovernorTimelockCompound(TIMELOCK)
+    Governor(GOVERNOR_NAME)
+  {}
+
+  /// @dev We override this function to resolve ambiguity between inherited contracts.
+  function supportsInterface(bytes4 interfaceId)
```

Comment:
Should `GovernorCountingFractional` have it's own ERC-165 `supportsInterface` definition so users can query it to determine if it supports fractional voting? It doesn't currently have this, but that could be useful (personally I've never gotten much value from ERC-165 and haven't heard of many people/tools who rely on it, feels like it might be nicer in theory than in practice)

Reply:
Good question! @apbendi What do you think

Reply:
I don't think so. ERC-165 is about the function signatures. FV does not change these. It's just about how we interpret the bytes passed to `castVoteWithReasonAndParams`.

For an integrator that wants to interrogate whether a given Governor supports fractional voting, they would use the value returned by [counting mode](https://github.com/ScopeLift/flexible-voting/blob/e5de2efd1368387b840931f19f3c184c85842761/src/GovernorCountingFractional.sol#L55), namely the `params=fractional`.

Reply:
> I don't think so. ERC-165 is about the function signatures. FV does not change these.

It does—ERC-165 defines "the interface identifier as the XOR of all function selectors in the interface", and FV v1.1.0 added the `fractionalVoteNonce` state var / getter method to the interface. I've created https://github.com/ScopeLift/flexible-voting/issues/53 to track this upstream

> For an integrator that wants to interrogate whether a given Governor supports fractional voting, they would use the value returned by [counting mode](https://github.com/ScopeLift/flexible-voting/blob/e5de2efd1368387b840931f19f3c184c85842761/src/GovernorCountingFractional.sol#L55), namely the `params=fractional`.

Good point here though, given this AFAICT there isn't much reason for integrator's to use ERC-165 with OZ governors

Reply:
Ahh yes, since the nonce was added as a bugfix, we missed this. I think this is a minimal issue that should not be considered a blocker for PT, given that, as you say, there's not a practical reason to use ERC-165 with the Governors. For correctness sake, we probably should issue a new patch version of FV that fixes it in the future, though. Thanks for creating the issue!